client/src/game/ui/HanabiCard.ts

```diff
@@ -375,15 +388,35 @@ export default class HanabiCard extends Konva.Group implements NodeWithTooltip {
       return UNKNOWN_CARD_RANK;
     }
 
-    // If we have a note identity on the card, show the rank corresponding to the note
-    // (specifically for stack bases in ongoing games; we want notes to have precedence in this case
-    // so that players can make notes in "Throw It in a Hole" variants)
+    // If we have a note on the card and it only provides possibilities of the same rank,
+    // show that rank (specifically for stack bases in ongoing games; we want notes to have
+    // precedence in this case so that players can make notes in "Throw It in a Hole" variants)
+
+    let noteRank = null;
+    if (this.note.possibilities.length !== 0) {
+      const possibleCardsFromNoteAndClues = this.note.possibilities.filter(
+        ([suitIndexA, rankA]) =>
+          this.state.possibleCardsFromClues.findIndex(
+            ([suitIndexB, rankB]) =>
+              suitIndexA === suitIndexB && rankA === rankB,
+          ) !== -1,
+      );
+      const [, candidateRank] = possibleCardsFromNoteAndClues[0];
```

Comment:
i'm not a fan of this notation, it seems confusing and hard to read
how about [0][1]

Reply:
I like it better. It's put a name to the variable rather than `[0]` which you'd have to find where the value is produced to understand.

Reply:
`const candidateRank = possibleCardsFromNoteAndClues[0][1];`

this is what i meant

Reply:
should I change this still or keep both the same?

Reply:
change please.
(don't you agree that having one index be on the left side and one index be on the right side is just bad code?)

Reply:
> change please.
K

> (don't you agree that having one index be on the left side and one index be on the right side is just bad code?)
No. The two indexes mean different things. One of them is an index into an array. The other one is essentially a field access on a tuple of (suitIndex, rank). I would rather access a field by name than by number.

Reply:
you can use array destructuring for all kinds of arrays, not just tuples.
so the line doesn't necessarily betray that the data structure is a tuple.
i guess your point is that by using array destructuring, you inject-into-the-code-style an implication that a tuple exists?
it just comes off to me like using arcane syntax for not a huge benefit

Reply:
In the language I'm used to, doing the destructuring would only be valid if the array was known to have exactly two elements and if that were the case in TS, then doing the destructuring would make it very clear to the reader that it is not in fact an array. Knowing that in TS you can destructure even if it has more elements, I don't really care. I'll just change both to indexing I suppose

Reply:
> In the language I'm used to, doing the destructuring would only be valid if the array was known to have exactly two elements
indeed, that sounds pretty useful. what language is that

Reply:
Rust

Reply:
and by the way, array destructuring is a JS feature not a TS feature